server/src/main/java/org/apache/druid/server/audit/SQLAuditManager.java

```diff
@@ -98,6 +98,10 @@ public void doAudit(AuditEntry auditEntry, Handle handle) throws IOException
             .setDimension("key", auditEntry.getKey())
             .setDimension("type", auditEntry.getType())
             .setDimension("author", auditEntry.getAuditInfo().getAuthor())
+            .setDimension("comment", auditEntry.getAuditInfo().getComment())
+            .setDimension("remote_address", auditEntry.getAuditInfo().getIp())
+            .setDimension("created_date", auditEntry.getAuditTime().toString())
+            .setDimension("payload", auditEntry.getPayload())
```

Comment:
Hmm, do we need to emit `comment` and `payload` as metrics? They seem to need to be retrieved via API instead.

Reply:
Typically `comment` tends to have the business reason for doing the change. So it is useful for auditing. I am on the fence about `payload`. I wasn't sure whether it has any extra information not covered by the other columns. So decided to add that one. I can remove it if that is not the case.

Reply:
Thanks. How about removing `payload` for now then? We can add it later if it turns out to be useful.

Reply:
Actually I checked some of the emitter metric logs and it does seem `payload` has some information not part of other columns. For example, for `audit_type` `rules` it contains the actual load rule change information, similarly for `coordinator.config` it contains the config setting. Looking at this I think I would like to keep the payload information around as well.

Reply:
I would like to make sure that it will be useful before we add any new dimension since it will increase the size of metrics. The metric size could be problematic since this method is used in updating lookups and compaction configurations. If it's not clear now, then we can add it later when it turns out to be useful.

Reply:
I have double checked, all the dimensions convey useful information. Which column is not clear to you?

Reply:
I'm concerned about the size of the payload. That is usually an unbounded json blob. So this could have scale implications. For example an audit message of a compaction change for a datasource in a cluster with 10k+ datasources can generate a payload that's several MB of text. 

If reporting payload is critical for some use cases, it should be disabled by default so users who want it can opt in.

Reply:
@suneet-s Interesting, is this because all the compaction config for 10K+ datasources get generated even when we are updating only 1? Do you think we can optimize that?

I am fine with making `payload` optional. My only concern is none of the dimension columns capture the actual change that happened which I believe is part of `payload`. This is important from the perspective of auditing.

Reply:
@mghosh4 I agree with you. I think it's a design flaw of how audit logging works in general, so we need to address that design holistically. Right now, even if I look at the `druid_audit` table, I need to do a json diff to actually know what change happened. I haven't created a proposal on how to fix that because I haven't had the time to come up with a meaningful solution to this, and come up with a testing / transition plan so that we don't break the existing audit logging system.

Reply:
Got it. So let me add a parameter called `addPayloadToAuditLogMetricDimension` in `SQLAuditManagerConfig`. I can use that to set the payload. Does that sound good? @suneet-s @jihoonson